### DISCUSSION TITLE
Replace Electron with Tauri

### DISCUSSION THREAD

**Discussion by alphaleadership** [4ian/GDevelop#6419] [Enhancement ideas (and their technical discussions)]
## Description

replace electron to tauri 

## Solution suggested

tauri create faster app than tauri

**Comment by AlexandreSi** [comment]
Hi @alphaleadership,

Could you answer theses questions before continuing further with this matter:
- Could you explain the reasons why you would like GDevelop to use Tauri instead of Electron?
- What would you like to be packaged with Tauri? the GDevelop app of the games made with GDevelop?
  - Depending on the answer, do you know if every Electron features used by GDevelop are available with Tauri?
- Do you have experience working with Tauri?

**Reply by alphaleadership** [reply]
tauri is faster and ligther and yes i work with tauri

**Reply by Silver-Streak** [reply]
Can you provide context and actual data/sources showing Tauri being faster?

Lighter, definitely, but every benchmark I can find shows no rendering performance difference for WebGL or WebGPU apps.

**Reply by alphaleadership** [reply]
dont use v8 engine need less ram an run faster is just logic

**Reply by Silver-Streak** [reply]
From some checking, it is only 40 mb difference of ram usage at idle and load. This much difference on a browser is not a real world performance impact.

"Run faster" doesn't mean anything without data. If you can provide benchmarks showing Tauri running WebGL apps with better performance than Electron, that'll be helpful for the dev team to make a decision on whether to build out an export path for it. 

Separately, as recently as May of last year Tauri's maintainers stated on their GitHub that they only support WebGPU via an unsafe/experimental flag in WebView2, which could mean that Tauri couldnt use many of the rendering features of PixiJS 8, which may make it unlikely to be an export target. Also, since Tauri uses system WebView equivalents on each operating system, rendering can (and will) be different per OS due to font rendering using the browser for standards. This would potentially be another risk from a game dev perspective.


Also, all of this may be moot since I just read that Tauri is rust based and not JS based, which may mean the devs would have to work in a completely different language if they are going to reimplement the Electron specific features into Tauri.

Edit: to be clear. None of the above means "they won't do it" or anything, just potential risks and reasons why it might not happen. If you can provide actual WebGL benchmark comparisons that will help the review process if the devs do look deeper.

**Reply by alphaleadership** [reply]
technicaly you can use backend function with windows.__tauri__
